book/src/pil/builtins.md

```diff
@@ -135,6 +138,20 @@ machine Sqrt {
 }}
 ```
 
+
+### Challenges
+
+```rust
+let std::prover::challenge: int, int -> expr
+```
+
+Constructs a challenge object, essentially asking the verifier for a random number.
+
+The first argument is the proof stage and the second is the identifier of the challenge.
+
+If you want two challanges to be different, you have to choose different IDs.
```

Comment:
```suggestion
If you want two challenges to be different, you have to choose different IDs.
```
IDE spellchecker?

Reply:
Can you recommend one? I tried out some, but they had so many false negatives.

Reply:
idk, vim has vim-spellchecker. It does have false negatives especially with technical words (ZK SNARKs) etc, but you can just ignored them or add to a dictionary